mail_gateway_whatsapp/models/mail_gateway.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import requests
+from werkzeug.urls import url_join
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+BASE_URL = "https://graph.facebook.com/"
+
+
+class MailGateway(models.Model):
+    _inherit = "mail.gateway"
+
+    whatsapp_security_key = fields.Char()
+    gateway_type = fields.Selection(
+        selection_add=[("whatsapp", "WhatsApp")], ondelete={"whatsapp": "cascade"}
+    )
+    whatsapp_from_phone = fields.Char()
+    whatsapp_version = fields.Char(default="15.0")
```

Comment:
The minimal API version supported by META for new apps is `23.0`. It could be a good time to update this as the default value. WDYT?
```suggestion
    whatsapp_version = fields.Char(default="23.0")
```